MathLib/TemplatePoint.h

```diff
@@ -92,6 +92,21 @@ template <typename T, std::size_t DIM = 3> class TemplatePoint
 		std::copy(std::istream_iterator<T>(is), std::istream_iterator<T>(), _x.begin());
 	}
 
+	/// operator -=
+	inline void operator-= (const TemplatePoint<T,DIM> &p)
```

Comment:
This operator is overloaded in `TemplateVector3` already. Maybe you can use objects of class `TemplateVector3` in your application?